reinstall-cli/src/users.rs

```diff
@@ -0,0 +1,106 @@
+use anyhow::{Context, Result};
+use std::fmt::Display;
+use std::fmt::Formatter;
+use uzers::os::unix::UserExt;
+
+#[derive(Clone, Debug)]
+pub(crate) struct UserKeys {
+    pub(crate) user: String,
+    pub(crate) authorized_keys: String,
+    pub(crate) authorized_keys_path: String,
+}
+
+impl UserKeys {
+    pub(crate) fn num_keys(&self) -> usize {
+        self.authorized_keys.lines().count()
+    }
+}
+
+impl Display for UserKeys {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "User {} ({} authorized keys)",
+            self.user,
+            self.num_keys()
+        )
+    }
+}
+
+/// # Safety
+/// See safety information in [`uzers::all_users`]. We are not impacted because we're not
+/// multi-threaded.
+pub(crate) fn get_all_users_keys() -> Result<Vec<UserKeys>> {
+    #[allow(unsafe_code)]
+    let iter = unsafe { uzers::all_users() };
+
+    let mut result = Vec::new();
+
+    for user_info in iter {
+        if user_info.uid() < 1000 && user_info.uid() != 0 {
```

Comment:
Note though 1000 is recommend, it's not *required* to be the barrier. See https://systemd.io/UIDS-GIDS/

But see above